### PR TITLE
spin up thread for goal tool to not block frontend

### PR DESCRIPTION
**Pull request by SteveMacenski**
https://github.com/ros-planning/navigation2/issues/1002

Makes it so RVIZ doesnt lag when you use the action server interface

### CODE REVIEW

File: nav2_rviz_plugins/src/goal_tool.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@ GoalTool::GoalTool()
 
 GoalTool::~GoalTool()
 {
+  execution_thread_->join();
```
Comment (orduno) [inline nav2_rviz_plugins/src/goal_tool.cpp]: `execution_thread_` has been detached, no need to `join`.

File: nav2_rviz_plugins/src/goal_tool.cpp [hunk 2]
```diff
@@ -45,6 +46,14 @@ void GoalTool::onInitialize()
 
 void
 GoalTool::onPoseSet(double x, double y, double theta)
+{
+  execution_thread_ = std::make_unique<std::thread>(&GoalTool::onPoseSetImpl, this,
```
Comment (orduno) [inline nav2_rviz_plugins/src/goal_tool.cpp]: maybe no need to have `execution_thread_` as a member variable as it's detaching

**Comment by bpwilcox** [comment]
@SteveMacenski  Will this PR be relevant once #944 is merged?

**Comment by SteveMacenski** [comment]
@bpwilcox I believe so, yes. As long as there's [stuff] that takes awhile (services, actions, wait for servers to be active) being called in the main thread of the plugin, then there's going to be blocking for the front end since its called by rviz in a single threaded fashion.

**Comment by SteveMacenski** [comment]
We could close this though if that's going to be merged soon and I can re-look at it. If its not going to be merged soon, I think its good to have in either way for the next N days from a UX perspective

**Pull request closed without merging**